lib/features/mailbox/data/datasource/mailbox_datasource.dart

```diff
@@ -1,7 +1,19 @@
 import 'package:jmap_dart_client/jmap/account_id.dart';
 import 'package:jmap_dart_client/jmap/core/properties/properties.dart';
+import 'package:jmap_dart_client/jmap/core/state.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/get/get_mailbox_response.dart';
 import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
+import 'package:tmail_ui_user/features/mailbox/data/model/mailbox_cache_response.dart';
+import 'package:tmail_ui_user/features/mailbox/data/model/mailbox_change_response.dart';
 
 abstract class MailboxDataSource {
-  Future<List<Mailbox>> getAllMailbox(AccountId accountId, {Properties? properties});
+  Future<GetMailboxResponse?> getAllMailbox(AccountId accountId, {Properties? properties});
+
+  Future<MailboxChangeResponse> getChanges(AccountId accountId, State sinceState);
+
+  Future<MailboxCacheResponse> getAllMailboxCache();
+
+  Future<MailboxChangeResponse> combineMailboxCache(MailboxChangeResponse mailboxChangeResponse, List<Mailbox> mailboxList);
+
+  Future<void> asyncUpdateCache(MailboxChangeResponse mailboxChangeResponse);
```

Comment:
Should be:
```
Future<void> update(MailboxChangeResponse mailboxChangeResponse);
```

Reply:
This is to update the cache. Should be clear here. One word update doesn't solve the problem at all. Too obscure

Reply:
For the generic of datasource: we should not care it is cache or remote, we only care that we have the `update` operation for the datasource. That's the idea

Reply:
I don't understand what you mean. Please give me details.

Reply:
Please refer: https://github.com/linagora/tmail-flutter/pull/127#discussion_r718213923